### DISCUSSION TITLE
Why two bech32 crates in this org?

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#1806] [Q&A] [answered]
Why do we have both
- https://github.com/rust-bitcoin/rust-bech32
- https://github.com/rust-bitcoin/rust-bech32-bitcoin

I thought bech32 was a Bitcoin thing, what does it mean to have `rust-bech32-bitcoin` _and_ `rust-bech32`. If I cannot reason about this and I work "inside" this org how are external users of our repos supposed to work out what is what?

**Comment by apoelstra** [comment, marked as answer]
Historically rust-bech32 was simply the error correcting code and encoding, while rust-bitcoin-bech32 had support for witness programs and other "bitcoin-specific" stuff. We eventually moved all that logic into the `Address` type of rust-bitcoin.

It looks like rust-bitcoin-bech32 is still maintained, which is a surprise to me, and is used in several miscellaneous utility crates. But it is not used by anything in the core rust-bitcoin ecosystem and I think it's fine if you and I forget that it exists.
